vm-linux-helpers.cmake

```diff
@@ -52,6 +52,46 @@ function(AddFileToOverlayDir filename file_location root_location overlay_name)
     )
 endfunction(AddFileToOverlayDir)
 
+# Function to add a symbolic link to an overlay directory.
+# filename: The name of the link in the overlay
+# file_location: Absolute path to the file we are adding the overlay
+# root_location: The folder in VM's root filesystem where the link is created in.
+# overlay_name: Overlay target name
+function(AddLinkToOverlayDir filename file_location root_location overlay_name)
+    # Get any existing dependencies when adding the file into the overlay
+    cmake_parse_arguments(PARSE_ARGV 4 ROOTFS_FILE_OVERLAY "" "" "DEPENDS")
+    if(ROOTFS_FILE_OVERLAY_UNPARSED_ARGUMENTS)
+        message(FATAL_ERROR "Unknown arguments to AddFileToOverlay")
+    endif()
+    set(rootfs_output_dir "${CMAKE_CURRENT_BINARY_DIR}/${overlay_name}")
```

Comment:
Could make this a bit more robust. Also, set property `ROOTFS_OUTPUT_DIR` when creating a new target instead of appending it.

```
if(NOT TARGET ${overlay_name})
        add_custom_target(${overlay_name})
        set_property(
            TARGET ${overlay_name}
            PROPERTY ROOTFS_OUTPUT_DIR "${CMAKE_CURRENT_BINARY_DIR}/${overlay_name}"
        )
endif 

get_target_property(rootfs_output_dir ${overlay_name} ROOTFS_OUTPUT_DIR)
if(NOT rootfs_output_dir)
    message(FATAL_ERROR "target '${overlay_name}' has no ROOTFS_OUTPUT_DIR")
endif()
```

Reply:
APPEND to an empty property works the same way as set so I don't agree it needs to be changed.

Checking that the property is set correctly on subsequent calls would make it more robust.  (This check should also be added to AddFileToOverlayDir) so it could be done for both in a follow up commit / PR.

Reply:
I agree that if a change should be done in both functions, it should be made in a followup PR